cypress/e2e/shared/explorer.test.ts

```diff
@@ -909,4 +910,41 @@ describe('DataExplorer', () => {
       })
     })
   })
+
+  describe('download csv', () => {
+    const downloadsDirectory = Cypress.config('downloadsFolder')
+
+    beforeEach(() => {
+      cy.writeData(points(20))
+      cy.task('deleteDownloads', {dirPath: downloadsDirectory})
+      cy.getByTestID('switch-to-script-editor').should('be.visible').click()
+    })
+
+    it('can download a file', () => {
+      cy.intercept('POST', '/api/v2/query?*').as('query')
+
+      cy.getByTestID('time-machine--bottom').within(() => {
+        cy.getByTestID('flux-editor', {timeout: 30000}).should('be.visible')
+          .monacoType(`from(bucket: "defbuck")
+  |> range(start: -10h)`)
+        cy.getByTestID('time-machine--download-csv')
+          .should('be.visible')
+          .click()
+      })
+
+      cy.wait('@query')
+      cy.wait(1200)
+
+      cy.task('getDownloads', {dirPath: downloadsDirectory}).should(files => {
+        ;(files as string[]).forEach(filename => {
```

Comment:
Does this semi-colon have some colloquial meaning?

Reply:
Semi-colon was automatically added by our linter.

There are reasons why a dev may choose to add this semi-colon themselves (e.g. as a sanity-check that post-javascript parsing, any previous 'bad' script has ended). But I didn't add it myself -- the linter automatically added it, and fails if I removed it.

Reply:
Can you file a bug about this? A linter shouldn't be making _less_ readable code.

Reply:
This isn't a bug, this is an example of Automatic Semicolon Insertion (ASI) from the compiler. This happens because the expression.

```ts
(files as string[])
```

could be interpreted as a function call if a line above it was added that looked like this:

```ts
const foo = bar()
```

which the compiler would interpret as:

```ts
const foo = bar()(files as string[])
```

[This doc](https://prettier.io/docs/en/rationale.html#semicolons) explains it better than I can.

I *think* a simple fix for this would be:

```ts
cy.task('getDownloads', {dirPath: downloadsDirectory}).should((files: string[]) => {
  files.forEach(filename =>
```

Reply:
Ah, so there is a spelling that makes that more clear and eliminates the ASI. Could you make those changes, @wiedld

Reply:
`files.forEach(` fails typechecking. Hence why I added the explicit casting...which then the linter required the ASI.